torch/csrc/jit/python/init.cpp

```diff
@@ -1700,6 +1700,13 @@ void initJITBindings(PyObject* module) {
           [](const FunctionSchema& self, const FunctionSchema& other) {
             return self == other;
           })
+      .def(
+          "__hash__",
+          [](const FunctionSchema& self) {
+            std::stringstream ss;
+            ss << self;
+            return std::hash<std::string>()(ss.str());
```

Comment:
This is a very inefficient way to go about hashing. More conventional way to do hashing is to inspect the equality function, hash each component and combine the hashes

Reply:
Thanks for the guide! Just updated the PR to use individual component hashing and combine them.